thewalrus/symplectic.py

```diff
@@ -67,7 +67,8 @@ def expand(S, modes, N):
         array: the resulting :math:`2N\times 2N` Symplectic matrix
     """
     M = len(S) // 2
-    S2 = np.identity(2 * N)
+    Sdtype = S.dtype
+    S2 = np.identity(2 * N, dtype=Sdtype)
```

Comment:
Why not just combine the two lines?

```suggestion
    S2 = np.identity(2 * N, dtype=S.dtype)
```

Reply:
I always find these thing confusing. Half the time people will tell you to split it for "readability" half the time I would just merge them for conciseness. I am fine either way